### PR TITLE
x86 self-hosted standalone object file headers

### PR DESCRIPTION
**Pull request by MadMax129**
I have begun working on the obj file headers for the self-hosted standalone compiler. For now, I have started the macho 32/64 bit headers but I will keep adding more. I have also added size tests.

### CODE REVIEW

**Comment by daurnimator** [comment]
The standard library already has macho definitions: https://github.com/ziglang/zig/blob/master/lib/std/macho.zig

**Comment by andrewrk** [comment]
Nice, can you merge this into the existing std lib definitions?

**Comment by MadMax129** [comment]
Just merged these changes into the std macho.zig file. Also added documentation alongside it.

File: lib/std/macho.zig [hunk 1]
```diff
@@ -268,6 +268,343 @@ pub const nlist_64 = extern struct {
     n_value: u64,
 };
 
+/// Defines the attributes of a LC_TWOLEVEL_HINTS load command
+pub const twolevel_hints_command = struct {
```
Comment (daurnimator) [inline lib/std/macho.zig:272]: should these all be `extern struct`?

Comment (daurnimator) [inline lib/std/macho.zig:272]: I know existing things in this file aren't; but should these be TitleCase to follow zig conventions?

File: lib/std/macho.zig [hunk 2]
```diff
@@ -268,6 +268,343 @@ pub const nlist_64 = extern struct {
...
+    /// Current version of the shared library
+    current_version: u32, 
+
+    /// Compatibility version of the shared library
+    compatibility_version: u32
+};
+
+/// Defines the attributes of the LC_LOAD_DYLIB and LC_ID_DYLIB load command
+pub const dylib_command = struct { 
+    /// Either LC_LOAD_DYLIB or LC_LOAD_WEAK_DYLIB or LC_ID_DYLIB
```
Comment (daurnimator) [inline lib/std/macho.zig:323]: Should this be an enum?:
```zig
pub const DylibCommand = extern enum(u32) {
  LOAD_DYLIB,
  LOAD_WEAK_DYLIB,
  ID_DYLIB,
}
```

similar for other cmd-like fields in this file.

Comment (MadMax129) [reply lib/std/macho.zig:323]: Most of these are already defined as const variables so I believe not, however we could move them into an enum if necessary

**Comment by MadMax129** [comment]
Well turns out the stuff depended on the snake style struct definitions meaning that since I changed them the debug.zig cant build. I can either fix the debug.zig or revert the changes any recommendations?

**Comment by andrewrk** [comment]
@MadMax129 sorry for the delay in reviewing this. My recommendation is to follow the conventions from the relevant macho .h files.

**Comment by andrewrk** [comment]
I think if you try this again you'll have better luck this time - @kubkon is also working on mach-o stuff and can help get your PRs landed

**Comment by kubkon** [comment]
Nice @MadMax129, sorry I missed this PR, somehow slipped under the radar. Anyhow, this looks cool and will definitely make my life easier with the Mach-O linker. See #6085 for some most recent work.

**Comment by kubkon** [comment]
Oh, and btw, do we even want to keep the 32bit structs at all given that, as far as I understand, we'll be dropping 32bit support in line with [Apple](https://support.apple.com/en-us/HT208436)? Maybe we should just drop them and keep one version only without the silly `64`/`64bit` suffix anywhere? Thoughts @andrewrk @daurnimator?

**Comment by andrewrk** [comment]
My proposal would be to drop 32 bit support unless someone shows up and wants to champion it

**Comment by kubkon** [comment]
@MadMax129 could you drop the 32bit support and rebase on top of `master`? You'll need to tweak [src-self-hosted/link/MachO.zig#L92](https://github.com/ziglang/zig/blob/master/src-self-hosted/link/MachO.zig#L92) with the new struct names, etc. though.

**Comment by andrewrk** [comment]
I'm sorry for the lack of engagement on this one. Sometimes I can't keep up with all the PRs and people are left hanging. You're welcome to pick this up again if you want to, Maks.

**Pull request closed without merging**